cmake/toolchain/xcc.cmake

```diff
@@ -1,6 +1,6 @@
 set(TOOLCHAIN_HOME /opt/xtensa/XtDevTools/install/tools/$ENV{TOOLCHAIN_VER}/XtensaTools)
 
-set(COMPILER gcc)
+set(COMPILER xcc)
```

Comment:
Doing this would be inconsistent with how other toolchains solve the same problem.

We should instead do as xcc did with 'CC' for instance.

Have gcc.cmake do

if(NOT DEFINED NOSYSDEF_CFLAG)
set(NOSYSDEF_CFLAG -undef)
endif()

and then do

set(NOSYSDEF_CFLAG "")

in toolchain/xcc.cmake

Reply:
Modified the patches to do this instead.